sklearn/cluster/_mean_shift.py

```diff
@@ -176,8 +176,10 @@ def mean_shift(
         operation terminates (for that seed point), if has not converged yet.
 
     n_jobs : int, default=None
-        The number of jobs to use for the computation. This works by computing
-        each of the n_init runs in parallel.
+        The number of jobs to use for the computation. The following tasks benefit
+        from the parallelization: the search of nearest neighbors for bandwidth
+        estimation and label assignments, and the hill climbing optimization
+        for all seeds.
```

Comment:
Perhaps this would improve readability, and provide a means of further reading.
```suggestion
        The number of jobs to use for the computation. The following tasks benefit
        from the parallelization:
        - The search of nearest neighbors for bandwidth estimation and label assignments
        - Hill-climbing optimization for all seeds
        
        See :term:`Glossary <n_jobs>` for more details.
```

Reply:
I also want your suggestion once, we can add it it looks good but this para will stand out because it is in points and para near it are like the real paragraphs. So if we don't have a problem with the look of the doc I think we can change it

Reply:
> I also want your suggestion on this, we can add it it looks good but this para will stand out because it is in points and para near it are like the real paragraphs. So if we don't have a problem with the look of the doc I think we can change it

Reply:
Actually, with the refactoring of the Cython code underlying the `NearestNeighbors`, `n_jobs` will soon have no effect on that step: the `n_jobs` parameter in `NearestNeighbors` is only used for very specific cases (e.g. distance metric passed as Python callable, which is never the case for neighbors computed for `MeanShift`).

Reply:
Hum it's true that it can be used for non-brute algorithms for small `n_features`. Maybe we can say:

> - The search of nearest neighbors for bandwidth estimation and label assignments. See the details in the docstring of the `NearestNeighbors` class.


to keep things simple in the `MeanShift` class.

Reply:
Yes I'll add it